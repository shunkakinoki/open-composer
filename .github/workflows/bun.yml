name: Bun
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  workflow_dispatch:
concurrency:
  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.head_ref || github.sha }}
  cancel-in-progress: true
jobs:
  bun-build:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Build
        run: bun run build
  bun-lint:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Check
        run: bun run check
      - name: Format
        run: bun run format
      - name: Lint
        run: bun run lint
      - name: Type Check
        run: bun run type-check
  bun-test:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Test
        run: bun run test
  bun-check:
    if: always()
    needs:
      - bun-build
      - bun-lint
      - bun-test
    runs-on: ubuntu-latest
    timeout-minutes: 3
    steps:
      - name: Alls Green
        uses: re-actors/alls-green@release/v1
        with:
          jobs: ${{ toJSON(needs) }}
