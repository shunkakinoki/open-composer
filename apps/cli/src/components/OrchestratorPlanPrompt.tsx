import { Box, Text, useApp, useInput } from "ink";
import type React from "react";
import { useState } from "react";

type Step =
  | "objective"
  | "description"
  | "constraints"
  | "tech-requirements"
  | "confirm";

interface OrchestratorPlanInitialState {
  step?: Step;
  objective?: string;
  description?: string;
  constraints?: string;
  techRequirements?: string;
}

interface OrchestratorPlanPromptProps {
  onComplete: (config: OrchestratorPlanConfig) => void;
  onCancel: () => void;
  initialState?: OrchestratorPlanInitialState;
}

export interface OrchestratorPlanConfig {
  objective: string;
  description: string;
  constraints?: string[];
  technicalRequirements?: string[];
}

export const OrchestratorPlanPrompt: React.FC<OrchestratorPlanPromptProps> = ({
  onComplete,
  onCancel,
  initialState,
}) => {
  const {
    step: initialStep = "objective",
    objective: initialObjective = "",
    description: initialDescription = "",
    constraints: initialConstraints = "",
    techRequirements: initialTechRequirements = "",
  } = initialState ?? {};

  const [step, setStep] = useState<Step>(initialStep);
  const [objective, setObjective] = useState(initialObjective);
  const [description, setDescription] = useState(initialDescription);
  const [constraints, setConstraints] = useState(initialConstraints);
  const [techRequirements, setTechRequirements] = useState(
    initialTechRequirements,
  );
  const { exit } = useApp();

  const handleComplete = () => {
    const config: OrchestratorPlanConfig = {
      objective: objective.trim(),
      description: description.trim(),
      constraints:
        constraints.trim().length > 0
          ? constraints.split(",").map((s) => s.trim())
          : undefined,
      technicalRequirements:
        techRequirements.trim().length > 0
          ? techRequirements.split(",").map((s) => s.trim())
          : undefined,
    };

    onComplete(config);
    exit();
  };

  useInput(
    (input, key) => {
      if (key.escape || (key.ctrl && input === "c")) {
        onCancel();
        exit();
        return;
      }

      switch (step) {
        case "objective":
          if (key.return) {
            if (objective.trim()) {
              setStep("description");
            }
          } else if (key.backspace || key.delete) {
            setObjective(objective.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setObjective(objective + input);
          }
          break;

        case "description":
          if (key.return) {
            if (description.trim()) {
              setStep("constraints");
            }
          } else if (key.backspace || key.delete) {
            setDescription(description.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setDescription(description + input);
          }
          break;

        case "constraints":
          if (key.return) {
            setStep("tech-requirements");
          } else if (key.backspace || key.delete) {
            setConstraints(constraints.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setConstraints(constraints + input);
          }
          break;

        case "tech-requirements":
          if (key.return) {
            setStep("confirm");
          } else if (key.backspace || key.delete) {
            setTechRequirements(techRequirements.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setTechRequirements(techRequirements + input);
          }
          break;

        case "confirm":
          if (key.return) {
            handleComplete();
          }
          break;
      }
    },
    { isActive: true },
  );

  const renderStep = () => {
    switch (step) {
      case "objective":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              ü§ñ Professor Oak Project Planner
            </Text>
            <Box marginTop={1}>
              <Text>Project objective or goal: </Text>
              <Text color="green">{objective}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">Press Enter to continue, Esc to cancel</Text>
            </Box>
          </Box>
        );

      case "description":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              üìù Project Description
            </Text>
            <Box marginTop={1}>
              <Text>Detailed description: </Text>
              <Text color="green">{description}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">Press Enter to continue, Esc to cancel</Text>
            </Box>
          </Box>
        );

      case "constraints":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              ‚ö†Ô∏è Constraints
            </Text>
            <Box marginTop={1}>
              <Text>Constraints (comma-separated, optional): </Text>
              <Text color="green">{constraints}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">
                Examples: TypeScript,React,Must have tests
              </Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">
                Press Enter to continue (or skip), Esc to cancel
              </Text>
            </Box>
          </Box>
        );

      case "tech-requirements":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              üîß Technical Requirements
            </Text>
            <Box marginTop={1}>
              <Text>Tech requirements (comma-separated, optional): </Text>
              <Text color="green">{techRequirements}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">Examples: PostgreSQL,REST API,Docker</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">
                Press Enter to continue (or skip), Esc to cancel
              </Text>
            </Box>
          </Box>
        );

      case "confirm":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              ‚úÖ Confirm Project Plan Request
            </Text>
            <Box marginTop={1}>
              <Text>
                Objective: <Text color="yellow">{objective.trim()}</Text>
              </Text>
            </Box>
            <Box marginTop={1}>
              <Text>
                Description: <Text color="yellow">{description.trim()}</Text>
              </Text>
            </Box>
            {constraints.trim() && (
              <Box marginTop={1}>
                <Text>
                  Constraints:{" "}
                  <Text color="yellow">
                    {constraints
                      .split(",")
                      .map((s) => s.trim())
                      .join(", ")}
                  </Text>
                </Text>
              </Box>
            )}
            {techRequirements.trim() && (
              <Box marginTop={1}>
                <Text>
                  Tech Requirements:{" "}
                  <Text color="yellow">
                    {techRequirements
                      .split(",")
                      .map((s) => s.trim())
                      .join(", ")}
                  </Text>
                </Text>
              </Box>
            )}
            <Box marginTop={2}>
              <Text color="gray">
                Press Enter to generate plan, Esc to cancel
              </Text>
            </Box>
          </Box>
        );
    }
  };

  return (
    <Box flexDirection="column" padding={2}>
      {renderStep()}
    </Box>
  );
};
