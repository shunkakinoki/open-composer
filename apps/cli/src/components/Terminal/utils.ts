/**
 * Terminal utility functions
 * Inspired by termact: https://github.com/MasterGordon/termact/blob/main/src/utils.ts
 */

import type { TerminalStyle } from "./types.js";
import {
  bg,
  blink,
  bold,
  doubleUnderline,
  fg,
  hidden,
  inverse,
  italic,
  strikethrough,
  underline,
} from "./ansi.js";

export const applyStyle = (style: TerminalStyle): string => {
  let result = "";
  if (style.fg) result += fg(style.fg);
  if (style.bg) result += bg(style.bg);
  if (style.bold) result += bold();
  if (style.italic) result += italic();
  if (style.underline) result += underline();
  if (style.blink) result += blink();
  if (style.reverse) result += inverse();
  if (style.hidden) result += hidden();
  if (style.strikethrough) result += strikethrough();
  if (style.doubleUnderline) result += doubleUnderline();
  return result;
};

export const wrapText = (text: string, width: number): string[] => {
  const lines: string[] = [];
  const words = text.split(" ");
  let currentLine = "";

  for (const word of words) {
    if (currentLine.length + word.length + 1 <= width) {
      currentLine += (currentLine ? " " : "") + word;
    } else {
      if (currentLine) lines.push(currentLine);
      currentLine = word;
    }
  }

  if (currentLine) lines.push(currentLine);
  return lines;
};

export const truncateText = (text: string, maxLength: number): string => {
  if (text.length <= maxLength) {
    return text;
  }
  return text.slice(0, maxLength - 3) + "...";
};

export const padText = (
  text: string,
  width: number,
  align: "left" | "center" | "right" = "left",
): string => {
  if (text.length >= width) {
    return text.slice(0, width);
  }

  const padding = width - text.length;

  switch (align) {
    case "center": {
      const leftPad = Math.floor(padding / 2);
      const rightPad = padding - leftPad;
      return " ".repeat(leftPad) + text + " ".repeat(rightPad);
    }
    case "right":
      return " ".repeat(padding) + text;
    default:
      return text + " ".repeat(padding);
  }
};

export const getNtscGrayscale = (r: number, g: number, b: number): number => {
  return r * 0.299 + g * 0.587 + b * 0.114;
};
