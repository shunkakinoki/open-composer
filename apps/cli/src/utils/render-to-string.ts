import { render as inkRender } from "ink";
import type { ReactElement } from "react";
import { EventEmitter } from "node:events";

/**
 * Minimal writable stream for capturing Ink output
 */
class CaptureStream extends EventEmitter {
  private output = "";

  write(chunk: string): void {
    this.output += chunk;
  }

  getOutput(): string {
    return this.output;
  }

  end(): void {
    // no-op
  }
}

/**
 * Renders a React component to a string using Ink
 * Useful for rendering UI components in Effect CLI commands
 */
export function renderToString(element: ReactElement): string {
  const stdout = new CaptureStream();
  const stderr = new CaptureStream();

  const instance = inkRender(element, {
    stdout: stdout as unknown as NodeJS.WriteStream,
    stderr: stderr as unknown as NodeJS.WriteStream,
    debug: false,
    exitOnCtrlC: false,
    patchConsole: false,
  });

  // Wait for a tick to let Ink render
  const output = stdout.getOutput();

  // Clean up
  instance.unmount();
  instance.cleanup();

  return output;
}
