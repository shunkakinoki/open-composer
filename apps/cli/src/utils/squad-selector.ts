import inquirer from "inquirer";
import type { LLMTask } from "@open-composer/agent-registry";
import { getAgentRegistry, getSquadLauncher } from "@open-composer/agent-registry";

/**
 * Start interactive squad selector
 */
export async function startSquadSelector(): Promise<void> {
  const registry = getAgentRegistry();
  const launcher = getSquadLauncher();

  console.log("\n🎮 Pokemon-Style Squad Manager\n");

  const { action } = await inquirer.prompt([
    {
      type: "list",
      name: "action",
      message: "What would you like to do?",
      choices: [
        { name: "📋 List all agents", value: "list" },
        { name: "✨ Create new squad", value: "create" },
        { name: "🚀 Quick squad (by task)", value: "quick" },
        { name: "📊 View statistics", value: "stats" },
        { name: "🚪 Exit", value: "exit" },
      ],
    },
  ]);

  switch (action) {
    case "list": {
      const agents = registry.getAllAgents();
      console.log(`\n📦 Found ${agents.length} agents:\n`);
      for (const agent of agents) {
        console.log(
          `  ${agent.pokemonAttributes.sprite} ${agent.name} (${agent.tier}) - ${agent.provider}`,
        );
      }
      break;
    }

    case "create": {
      console.log("\n🛠️  Create Squad - Not implemented in CLI yet");
      console.log("   Use the interactive UI components for full squad creation");
      break;
    }

    case "quick": {
      const { task } = await inquirer.prompt([
        {
          type: "list",
          name: "task",
          message: "Select task specialization:",
          choices: [
            { name: "🧭 Planner", value: "planner" },
            { name: "⚡ Programmer", value: "programmer" },
            { name: "🔍 Reviewer", value: "reviewer" },
            { name: "🌊 Router", value: "router" },
            { name: "🌿 Summarizer", value: "summarizer" },
          ],
        },
      ]);

      const team = launcher.createQuickSquad(task as LLMTask);
      console.log(`\n✅ Created squad: ${team.name}`);
      console.log(`   Agents: ${team.agents.map((a) => a.name).join(", ")}`);
      break;
    }

    case "stats": {
      const stats = registry.getStats();
      console.log("\n📊 Registry Statistics:");
      console.log(`   Total Agents: ${stats.totalAgents}`);
      console.log(`   Total Squads: ${stats.totalTeams}`);
      console.log(`   Starter: ${stats.tierCounts.starter || 0}`);
      console.log(`   Evolved: ${stats.tierCounts.evolved || 0}`);
      console.log(`   Legendary: ${stats.tierCounts.legendary || 0}`);
      break;
    }

    case "exit":
      console.log("\n👋 Goodbye!\n");
      return;
  }

  // Loop back to main menu
  await startSquadSelector();
}
