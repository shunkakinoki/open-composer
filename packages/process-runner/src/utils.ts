import * as Effect from "effect/Effect";
import {
  MAX_COMMAND_LENGTH,
  MAX_RUN_NAME_LENGTH,
  RUN_NAME_PATTERN,
} from "./constants.js";
import { ProcessRunnerError } from "./types.js";

// =============================================================================
// VALIDATION UTILITIES
// =============================================================================

export function validateRunName(
  runName: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!runName || typeof runName !== "string") {
    return Effect.fail(
      ProcessRunnerError("Run name must be a non-empty string"),
    );
  }

  if (runName.length > MAX_RUN_NAME_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Run name too long (max ${MAX_RUN_NAME_LENGTH} characters)`,
      ),
    );
  }

  if (!RUN_NAME_PATTERN.test(runName)) {
    return Effect.fail(
      ProcessRunnerError(
        "Run name can only contain letters, numbers, hyphens, and underscores",
      ),
    );
  }

  return Effect.succeed(runName);
}

export function validateCommand(
  command: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!command || typeof command !== "string") {
    return Effect.fail(
      ProcessRunnerError("Command must be a non-empty string"),
    );
  }

  if (command.length > MAX_COMMAND_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Command too long (max ${MAX_COMMAND_LENGTH} characters)`,
      ),
    );
  }

  // Basic command injection prevention - only reject null bytes
  if (command.includes("\x00")) {
    return Effect.fail(
      ProcessRunnerError("Command contains invalid characters"),
    );
  }

  return Effect.succeed(command);
}

// =============================================================================
// UTILITY FUNCTIONS
// =============================================================================

export function withTimeout<T>(
  effect: Effect.Effect<T, ProcessRunnerError>,
  timeoutMs: number,
  timeoutMessage: string = "Operation timed out",
): Effect.Effect<T, ProcessRunnerError> {
  const timeoutEffect = Effect.async<T, ProcessRunnerError>((resume) => {
    const timeout = setTimeout(() => {
      resume(Effect.fail(ProcessRunnerError(timeoutMessage)));
    }, timeoutMs);
    return Effect.sync(() => clearTimeout(timeout));
  });

  return Effect.race(effect, timeoutEffect);
}
