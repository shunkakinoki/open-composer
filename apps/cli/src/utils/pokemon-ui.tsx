import type { AgentTeam, OpenComposerAgent } from "@open-composer/agent-registry";
import type { RegistryStats } from "@open-composer/analytics";
import { renderToString } from "./render-to-string.js";
import {
  AgentCard,
  AgentComparison,
  AgentList,
  EvolutionTree,
  RegistryStats as RegistryStatsComponent,
  TaskSpecializations,
  TeamDisplay,
} from "../components/PokemonUI.js";

/**
 * Pokemon-style UI renderer using React components
 * This is a utility wrapper that renders React components to strings for CLI output
 */
export class PokemonUI {
  /**
   * Render agent card
   */
  static renderAgentCard(agent: OpenComposerAgent): string {
    return renderToString(<AgentCard agent={agent} />);
  }

  /**
   * Render agent list
   */
  static renderAgentList(agents: OpenComposerAgent[]): string {
    return renderToString(<AgentList agents={agents} />);
  }

  /**
   * Render team display
   */
  static renderTeam(team: AgentTeam): string {
    return renderToString(<TeamDisplay team={team} />);
  }

  /**
   * Render evolution tree
   */
  static renderEvolutionTree(agent: OpenComposerAgent): string {
    return renderToString(<EvolutionTree agent={agent} />);
  }

  /**
   * Render task specializations
   */
  static renderTaskSpecializations(agent: OpenComposerAgent): string {
    return renderToString(<TaskSpecializations agent={agent} />);
  }

  /**
   * Render registry statistics
   */
  static renderStats(stats: RegistryStats): string {
    return renderToString(<RegistryStatsComponent stats={stats} />);
  }

  /**
   * Render agent comparison
   */
  static renderComparison(
    agent1: OpenComposerAgent,
    agent2: OpenComposerAgent,
  ): string {
    return renderToString(<AgentComparison agent1={agent1} agent2={agent2} />);
  }
}
