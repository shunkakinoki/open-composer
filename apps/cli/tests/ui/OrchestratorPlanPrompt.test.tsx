import { describe, expect, mock, test } from "bun:test";

import { OrchestratorPlanPrompt } from "../../src/components/OrchestratorPlanPrompt.js";
import { render } from "../utils.js";

describe("OrchestratorPlanPrompt", () => {
  test("renders objective step snapshot by default", () => {
    const onComplete = mock(() => {});
    const onCancel = mock(() => {});

    const { lastFrame } = render(
      <OrchestratorPlanPrompt onComplete={onComplete} onCancel={onCancel} />,
    );

    expect(lastFrame()).toMatchSnapshot();
  });

  test("renders description step snapshot", () => {
    const onComplete = mock(() => {});
    const onCancel = mock(() => {});

    const { lastFrame } = render(
      <OrchestratorPlanPrompt
        onComplete={onComplete}
        onCancel={onCancel}
        initialState={{
          step: "description",
          objective: "Launch new CLI tool",
          description: "Plan execution strategy",
        }}
      />,
    );

    expect(lastFrame()).toMatchSnapshot();
  });

  test("renders constraints step snapshot", () => {
    const onComplete = mock(() => {});
    const onCancel = mock(() => {});

    const { lastFrame } = render(
      <OrchestratorPlanPrompt
        onComplete={onComplete}
        onCancel={onCancel}
        initialState={{
          step: "constraints",
          objective: "Launch new CLI tool",
          description: "Plan execution strategy",
          constraints: "TypeScript,React,Must include docs",
        }}
      />,
    );

    expect(lastFrame()).toMatchSnapshot();
  });

  test("renders tech requirements step snapshot", () => {
    const onComplete = mock(() => {});
    const onCancel = mock(() => {});

    const { lastFrame } = render(
      <OrchestratorPlanPrompt
        onComplete={onComplete}
        onCancel={onCancel}
        initialState={{
          step: "tech-requirements",
          objective: "Launch new CLI tool",
          description: "Plan execution strategy",
          constraints: "TypeScript,React,Must include docs",
          techRequirements: "PostgreSQL,REST API,Docker",
        }}
      />,
    );

    expect(lastFrame()).toMatchSnapshot();
  });

  test("renders confirm step snapshot", () => {
    const onComplete = mock(() => {});
    const onCancel = mock(() => {});

    const { lastFrame } = render(
      <OrchestratorPlanPrompt
        onComplete={onComplete}
        onCancel={onCancel}
        initialState={{
          step: "confirm",
          objective: "Launch new CLI tool",
          description: "Plan execution strategy",
          constraints: "TypeScript,React,Must include docs",
          techRequirements: "PostgreSQL,REST API,Docker",
        }}
      />,
    );

    expect(lastFrame()).toMatchSnapshot();
  });
});
