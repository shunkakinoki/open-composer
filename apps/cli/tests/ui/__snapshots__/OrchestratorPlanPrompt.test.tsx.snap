// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`OrchestratorPlanPrompt renders objective step snapshot by default 1`] = `
"

  ü§ñ Professor Oak Project Planner

  Project objective or goal: _

  Press Enter to continue, Esc to cancel

"
`;

exports[`OrchestratorPlanPrompt renders description step snapshot 1`] = `
"

  üìù Project Description

  Detailed description: Plan execution strategy_

  Press Enter to continue, Esc to cancel

"
`;

exports[`OrchestratorPlanPrompt renders constraints step snapshot 1`] = `
"

  ‚ö†Ô∏è Constraints

  Constraints (comma-separated, optional): TypeScript,React,Must include docs_

  Examples: TypeScript,React,Must have tests

  Press Enter to continue (or skip), Esc to cancel

"
`;

exports[`OrchestratorPlanPrompt renders tech requirements step snapshot 1`] = `
"

  üîß Technical Requirements

  Tech requirements (comma-separated, optional): PostgreSQL,REST API,Docker_

  Examples: PostgreSQL,REST API,Docker

  Press Enter to continue (or skip), Esc to cancel

"
`;

exports[`OrchestratorPlanPrompt renders confirm step snapshot 1`] = `
"

  ‚úÖ Confirm Project Plan Request

  Objective: Launch new CLI tool

  Description: Plan execution strategy

  Constraints: TypeScript, React, Must include docs

  Tech Requirements: PostgreSQL, REST API, Docker


  Press Enter to generate plan, Esc to cancel

"
`;
